src/components/forms/AttributeForm.jsx

```diff
@@ -0,0 +1,182 @@
+import { makeStyles } from '@material-ui/core/styles';
+import { useTranslation } from 'react-i18next';
+import StringField from '../input/StringField';
+import BoolField from '../input/BoolField';
+import DateTimeField from '../input/DateTimeField';
+import IntegerField from '../input/IntField';
+import FloatField from '../input/FloatField';
+import { IconButton, InputAdornment, Tooltip } from '@material-ui/core';
+import { Clear as ClearIcon } from '@material-ui/icons';
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+import CardHeader from '@material-ui/core/CardHeader';
+import Key from '@material-ui/icons/VpnKey';
+import Attributes from '@material-ui/icons/HdrWeakTwoTone';
+import Typography from '@material-ui/core/Typography';
+import Grid from '@material-ui/core/Grid';
+const useStyles = makeStyles((theme) => ({
+  root: {
+    margin: '0rem',
+  },
+  card: {
+    margin: '0rem',
+    maxHeight: '70vh',
+    overflow: 'auto',
+  },
+  cardB: {
+    margin: '0rem',
+    padding: '0rem',
+  },
+  cardContent: {
+    marginLeft: '5rem',
+    marginRight: '5rem',
+    minWidth: 200,
+  },
+}));
+
+const field = {
+  String: StringField,
+  Boolean: BoolField,
+  DateTime: DateTimeField,
+  Integer: IntegerField,
+  Float: FloatField,
+};
+
+export default function AttributeForm({ attributes, setAttributes, ...props }) {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const handleSetValue = (attrName, value) => {
+    setAttributes({ ...attributes, [attrName]: value });
+  };
```

Comment:
Just a minor observation, because I also looked at a similar implementation the other day. The `attrName` (label prop) is known by the form, so the handler could easily be refactored to be a higher-order function.

```js
const handleSetValue = (attrName) => (value) => {
    setAttributes({ ...attributes, [attrName]: value });
 };
```

Then in the input field:

```js
<StringField
      label={attrName}
      onChange={handleSetValue(attrName)}
/>
```

Though this probably means also doing a minor refactor on the input fields, so only the value is passed to the `onChange` callback.